src/main/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialect.java

```diff
@@ -53,17 +52,26 @@ public Capabilities getCapabilities() {
     @Override
     protected QueryRewriter createQueryRewriter() {
         try {
-            final TableProtectionStatus tableProtectionStatus = new TableProtectionStatus(
-                    this.connection.getMetaData());
-            return new RowLevelSecurityQueryRewriter(this, this.remoteMetadataReader, this.connection,
-                    tableProtectionStatus);
+            final String catalogName = this.properties.getCatalogName();
+            final String schemaName = this.properties.getSchemaName();
+            final DatabaseMetaData metadata = this.connectionFactory.getConnection().getMetaData();
+            final TableProtectionStatus tableProtectionStatus = new TableProtectionStatusReader(metadata)
+                    .read(catalogName, schemaName);
+            final QueryRewriter delegate = super.createQueryRewriter();
+            return new RowLevelSecurityQueryRewriter(this, createRemoteMetadataReader(), this.connectionFactory,
+                    tableProtectionStatus, delegate);
```

Comment:
Why is it named delegate? I was puzzled what it can be and it was QueryRewriter

Reply:
The RLS Rewriter does its part and leave the remaining Exasol-specifics to its delegate, the `ExasolQueryRewriter`.